pandas/core/index.py

```diff
@@ -1219,14 +1219,17 @@ def union(self, other):
         if len(other) == 0 or self.equals(other):
             return self
 
+        other = _ensure_index(other)
+
```

Comment:
The reason why i changed this is, that a union between an index and a numpy/python array right now crashes on AttributeErrors

index = pd.Index([1,2,3])
index.union([4,5,6])

-> 1238         if self.dtype != other.dtype:
   1239             this = self.astype('O')
   1240             other = other.astype('O')

AttributeError: 'list' object has no attribute 'dtype'

so I think the _ensure_index(other) has to be done regardless of the if len(self) == 0:

I also think it is implemented like that in intersection